torch/csrc/jit/passes/onnx/onnx_log.cpp

```diff
@@ -0,0 +1,29 @@
+#include <torch/csrc/jit/passes/onnx/onnx_log.h>
+#include <iostream>
+
+namespace torch {
+namespace jit {
+
+static bool onnx_log_enabled = false;
+static std::ostream* out = &std::cout;
+
+bool is_onnx_log_enabled() {
+  return onnx_log_enabled;
+}
+
+void set_onnx_log_enabled(bool enabled) {
+  onnx_log_enabled = enabled;
+}
+
+void set_onnx_log_output_stream(std::ostream* out_stream) {
+  if (nullptr != out_stream) {
+    out = out_stream;
+  }
+}
```

Comment:
Seen raw pointers in C++ code in 2022 is a bit weird. Who guarantees the lifetime of the object?
Let's use `shared_ptr` to ensure that out is valid unless somebody resets it
```suggestion
void set_onnx_log_output_stream(std::shared_ptr<std::ostream> out_stream) {
    out = std::move(out_stream);
}
```